project/sbt-changeling/Changeling.scala

```diff
@@ -0,0 +1,280 @@
+package systems.informal.sbt.changeling
+
+/**
+ * =Changeling=
+ *
+ * Changeling is an SBT pluging to make allow simple maintenance of a merge-conflict free changelog.
+ *
+ * ==Motivation==
+ *
+ * Merge conflicts arise when the same lines in the same files are edited
+ *
+ * "simultaneously" in two different branches. The basic idea is to reduce the chance of merge conflicts by adding each
+ * entry of an unreleased change into the changelog via a separate file.
+ *
+ * ===Design===
+ *
+ * In order to produce a changelog that has the usual and expected format in a flatfile, we just need a representation
+ * in the file system that will let us construct the expected result. If we look at a changelog, we can see an evident
+ * document model. Our changelog has roughly the following general shape:
+ *
+ * {{{
+ * # Changelog
+ *
+ * ## Unrelased
+ *
+ * ### Features
+ *
+ * - Change entry 1
+ * - Change entry 2
+ *
+ * ### Bug fixes
+ *
+ * - Change entry 3
+ * - Change entry 4
+ *
+ * ## 0.0.1
+ *
+ * ### Features
+ *
+ * - Change entry 1
+ * - Change entry 2
+ *
+ * ### Bug fixes
+ *
+ * - Change entry 3
+ * - Change entry 4
+ * }}}
+ *
+ * The obvious, high-level document model is a tree of depth 4:
```

Comment:
I've only heard "depth" applied to individual nodes, I think the entire tree has a height.
Also, I believe the convention is to count the path length, not the vertices.

```suggestion
 * The obvious, high-level document model is a tree of height 3:
```